pkg/apis/pipeline/v1beta1/run_interface.go

```diff
@@ -23,7 +23,7 @@ import (
 	"knative.dev/pkg/apis"
 )
 
-// RunObject is implemented by CustomRun and Run
+// RunObject is implemented by Run, CustomRun, TaskRun and PipelineRun
```

Comment:
`RunObject` was introduced for remote resolution but it's not used anymore anywhere.
It is still very useful for the events controller though, except for the `GetRetryCount`, which I moved to a different interface.